app/src/main/java/org/pragyan/dalal18/fragment/OTPVerificationDialogFragment.kt

```diff
@@ -0,0 +1,131 @@
+package org.pragyan.dalal18.fragment
+
+import android.content.Intent
+import android.content.SharedPreferences
+import android.graphics.Color
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.lifecycleScope
+import com.google.android.material.snackbar.Snackbar
+import dalalstreet.api.DalalActionServiceGrpc
+import dalalstreet.api.actions.LoginRequest
+import dalalstreet.api.actions.LoginResponse
+import dalalstreet.api.actions.VerifyOTPRequest
+import dalalstreet.api.actions.VerifyOTPResponse
+import io.grpc.ManagedChannel
+import kotlinx.android.synthetic.main.activity_login.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.jetbrains.anko.contentView
+import org.jetbrains.anko.doAsync
+import org.jetbrains.anko.toast
+import org.jetbrains.anko.uiThread
+import org.pragyan.dalal18.R
+import org.pragyan.dalal18.dagger.ContextModule
+import org.pragyan.dalal18.dagger.DaggerDalalStreetApplicationComponent
+import org.pragyan.dalal18.data.GlobalStockDetails
+import org.pragyan.dalal18.data.StockDetails
+import org.pragyan.dalal18.ui.LoginActivity
+import org.pragyan.dalal18.ui.MainActivity
+import org.pragyan.dalal18.ui.OtpEditText
+import org.pragyan.dalal18.ui.VerifyPhoneActivity
+import org.pragyan.dalal18.utils.ConnectionUtils
+import org.pragyan.dalal18.utils.Constants
+import org.pragyan.dalal18.utils.MiscellaneousUtils
+import org.pragyan.dalal18.utils.hideKeyboard
+import java.util.ArrayList
+import javax.inject.Inject
+
+class OTPVerificationDialogFragment : DialogFragment() {
+
+    @Inject
+    lateinit var channel: ManagedChannel
```

Comment:
Are you doing this because you dont't expect session ID to be attached as header when you inject this?